MosipNexus/crawler/web_crawler.py

```
@@ -0,0 +1,210 @@
+
+sys.path.insert(0, str(Path(__file__).parent.parent))
+from config.settings import (
+    HTTP_HEADERS,
+    CRAWL_DELAY_SECS,
+)
+
+_SM_NS = {"sm": "http://www.sitemaps.org/schemas/sitemap/0.9"}
+
+
+def _get_locs(xml_text: str) -> list[str]:
+    root = ET.fromstring(xml_text)
```

Ruff (0.15.20)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)